src/connectStoreController.ts

```diff
@@ -1,126 +1,89 @@
 import {
   PackageResponse,
-  RequestPackageFunction,
   RequestPackageOptions,
-  Resolution,
   WantedDependency,
 } from '@pnpm/package-requester'
-import JsonSocket = require('json-socket')
-import net = require('net')
+
+import got = require('got')
+import pLimit = require('p-limit')
 import {StoreController} from 'package-store'
 import uuid = require('uuid')
 
 export default function (
-  initOpts: object,
+  initOpts: {
+    path?: string,
+    port?: number,
+    hostname?: string,
+  },
 ): Promise<StoreController> {
-  const socket = new JsonSocket(new net.Socket());
-  socket.connect(initOpts as any) // tslint:disable-line
+  let remotePrefix: string
+  if (initOpts.path) {
+    remotePrefix = `http://unix:${initOpts.path}:`
+  } else {
+    remotePrefix = `http://${initOpts.hostname}:${initOpts.port}`
```

Comment:
I think it would be fine to pass in the formed path, these manipulations can be done by pnpm